File: py_java_scala_benchmark/fib.scala
object flib2 extends App {

  @scala.annotation.tailrec
  def flib(n: Int, preResult: Int, result: Int): Int = {
    if (n <= 1) result
    else flib(n - 1, result, preResult + result)
  }

  println(flib(36, 0, 1))
}

File: py_java_scala_benchmark/fib.java
public class fib {
    public static int flib(int n) {
        if (n < 2)
            return n;
        else
            return flib(n - 1) + flib(n - 2);
    }

    public static void main(String[] args) {
        System.out.println(flib(36));
    }
}
